larndsim/detsim.py

```diff
@@ -497,23 +504,26 @@ def sum_pixel_signals(pixels_signals, signals, track_starts, pixel_index_map, tr
         start_tick = round(track_starts[itrk] / detector.TIME_SAMPLING)
 
         if pixel_index >= 0:
-            counter = 0
+            counter = -99
             for track_idx in range(track_pixel_map[pixel_index].shape[0]):
-                if itrk == -1:
+                if itrk == -1: # would itrk ever be -1?
                     continue
                 if itrk == int(track_pixel_map[pixel_index][track_idx]):
                     counter = track_idx
+                    if counter >= 0 and itick < signals.shape[2]:
```

Comment:
It looks like this is the crucial difference in this PR (apart from changing `MAX_TRACKS_PER_PIXEL` and `MAX_ADC_VALUES`). Am I right in thinking that not having the `counter >= 0` in this if statement was causing the backtracking errors seen before? And did including this catch expose that we needed to increase the MAX values?

Reply:
This set of slides should serve as a visual aid for this problem and a validation: https://indico.fnal.gov/event/64745/contributions/291455/attachments/177843/242338/2024_05_14_larndsim_debug_MAX_TRACKS_PER_PIXEL.pdf

Reply:
The problem is a combination of the default `counter` value and where to fill `pixels_signals` and `pixels_tracks_signals`. Regardless, it's dangerous to set the default value to something physical. 0 and -1 (I think) both can be valid values.

Reply:
Isn't `track_pixel_map[pixel_index]` supposed to be guaranteed to contain `itrk`, since `pixel_index` comes from `pixel_index_map[itrk]`? As long as this holds, shouldn't `counter` always get set properly, so that the default value doesn't matter?

Reply:
@mjkramer `track_pixel_map[pixel_index]` does not guarantee to contain `itrk`. See the added comments. track_pixel_map.shape[1] goes up to `MAX_TRACKS_PER_PIXEL`, while signals include contribution from all possible segments. For example, if you set `MAX_TRACKS_PER_PIXEL` to 20 and you actually have 25 segments contributing to this pixel. There are some `itrk` value wouldn't find anything in this loop and `counter` remains 0, so the last 5 contributions (not ranked by size) are added to the 0th.
May the pictures here would help
https://indico.fnal.gov/event/64745/contributions/291455/attachments/177843/242338/2024_05_14_larndsim_debug_MAX_TRACKS_PER_PIXEL.pdf

Reply:
Got it, thanks.